crates/context_aware_config/src/api/config_versions/handlers.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{
+    db::{models::ConfigVersion, schema::config_versions},
+    types::QueryFilters,
+};
+
+extern crate base64;
+use service_utils::service::types::DbConnection;
+
+use superposition_types::result as superposition;
+
+use actix_web::{get, web::Query, HttpResponse, Scope};
+use diesel::{ExpressionMethods, QueryDsl, RunQueryDsl};
+use serde_json::json;
+
+pub fn endpoints() -> Scope {
+    Scope::new("").service(get)
+}
+
+#[get("")]
+async fn get(
+    db_conn: DbConnection,
+    filters: Query<QueryFilters>,
+) -> superposition::Result<HttpResponse> {
```

Comment:
Instead of HttpResponse , Can you change it to a type as SnapshotResponse or something?

Reply:
Yeah can be done but the response now is 
`"total_pages": total_pages,
    "total_items": n_version,
    "data": config_versions`
    but as pagination is there for so many APIs
    Maybe something like this can be done
    `enum DataType<A, B, C> {
    TypeA(Vec<A>),
    TypeB(Vec<B>),
    TypeC(Vec<C>),
}

struct PaginatedResponse<A, B, C> {
    total_pages: i64,
    total_items: i64,
    data: DataType<A, B, C>,
}`

Opinions?

Reply:
Yeah , that looks a better generic solution to me , we can do that.

Reply:
@mahatoankitkumar you don't need an enum for this
the PaginationResponse should be a generic type capable of returning any paginated response for any data type, an enum should be needed for it, as in case of Enum it would lead to unnecessary boilerplate code as well

simply doing this should be enough:
```
pub struct PaginatedResponse<T> {
      pub total_pages: i64,
      pub total_items: i64,
      data: T,
}
```

Reply:
One more addition to this, T should be deserializable, which can be done by `struct PaginatedResponse<T: DeserializeOwned>`.